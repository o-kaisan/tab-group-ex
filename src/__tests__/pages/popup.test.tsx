/**
 * @jest-environment jsdom
 *
 * TODO: メニューにテキストが表示されているか
 * TODO: onClickに渡した関数が呼ばれるか
 *
 */

 import React from 'react';
 import ReactDOM from "react-dom";
 import renderer from 'react-test-renderer'
 import PopupMenu from '../../pages/popup';

 describe('PopupMenu-test', () => {
     test('menu', () =>{
     // ToDo: ポップアップメニューのsnapshotをとって差分が見れるように
     // FIXME: 「Target container is not a DOM element.」テストできない。
    //  const tree = renderer
    //      .create(<PopupMenu/>)
    //      .toJSON();
    //  expect(tree).toMatchSnapshot();
     });
 });
