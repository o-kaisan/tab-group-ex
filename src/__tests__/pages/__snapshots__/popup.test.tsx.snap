// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupMenu-test menu 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-19v6clq-MuiPaper-root"
>
  <ul
    className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    onKeyDown={[Function]}
    role="menu"
    tabIndex={-1}
  >
    <li
      className="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="LayersIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m11.99 18.54-7.37-5.73L3 14.07l9 7 9-7-1.63-1.27-7.38 5.74zM12 16l7.36-5.73L21 9l-9-7-9 7 1.63 1.27L12 16z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          タブを全てグループ化
        </span>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        ⌘X
      </p>
    </li>
    <li
      className="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      <div
        className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="LayersClearIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m19.81 14.99 1.19-.92-1.43-1.43-1.19.92 1.43 1.43zm-.45-4.72L21 9l-9-7-2.91 2.27 7.87 7.88 2.4-1.88zM3.27 1 2 2.27l4.22 4.22L3 9l1.63 1.27L12 16l2.1-1.63 1.43 1.43L12 18.54l-7.37-5.73L3 14.07l9 7 4.95-3.85L20.73 21 22 19.73 3.27 1z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          タブのグループ化を解除
        </span>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        ⌘C
      </p>
    </li>
  </ul>
</div>
`;

exports[`PopupMenu-test メニューの表示内容が正しいことのテスト 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-19v6clq-MuiPaper-root"
>
  <ul
    className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    onKeyDown={[Function]}
    role="menu"
    tabIndex={-1}
  >
    <li
      className="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="LayersIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m11.99 18.54-7.37-5.73L3 14.07l9 7 9-7-1.63-1.27-7.38 5.74zM12 16l7.36-5.73L21 9l-9-7-9 7 1.63 1.27L12 16z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          タブを全てグループ化
        </span>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        ⌘X
      </p>
    </li>
    <li
      className="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      <div
        className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="LayersClearIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m19.81 14.99 1.19-.92-1.43-1.43-1.19.92 1.43 1.43zm-.45-4.72L21 9l-9-7-2.91 2.27 7.87 7.88 2.4-1.88zM3.27 1 2 2.27l4.22 4.22L3 9l1.63 1.27L12 16l2.1-1.63 1.43 1.43L12 18.54l-7.37-5.73L3 14.07l9 7 4.95-3.85L20.73 21 22 19.73 3.27 1z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          タブのグループ化を解除
        </span>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        ⌘C
      </p>
    </li>
  </ul>
</div>
`;
